src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -0,0 +1,557 @@
+        {
+            const auto oldKeyPair{ _KeyMap.find(keys) };
+            if (oldKeyPair != _KeyMap.end())
+            {
+                // Collision: The key chord was already in use.
+                // Remove the old one.
+                const auto& actionPair{ _ActionMap.find(oldKeyPair->second) };
+                const auto& conflictingCmd{ actionPair->second };
+                const auto& conflictingCmdImpl{ get_self<implementation::Command>(conflictingCmd) };
+                conflictingCmdImpl->EraseKey(keys);
+            }
+            else if (const auto& conflictingCmd{ GetActionByKeyChord(keys) })
+            {
+                // Collision: The key chord was already in use, but by an action in another layer
+                const auto conflictingActionID{ ActionHash{}(conflictingCmd.ActionAndArgs()) };
+                const auto& consolidatedCmdPair{ _ConsolidatedActions.find(conflictingActionID) };
+                if (consolidatedCmdPair != _ConsolidatedActions.end())
+                {
+                    const auto& consolidatedCmdImpl{ get_self<implementation::Command>(consolidatedCmdPair->second) };
+                    consolidatedCmdImpl->EraseKey(keys);
+                }
+                else
+                {
+                    // Create a copy of the conflicting action,
+                    // and erase the conflicting key chord from the copy.
+                    const auto& conflictingCmdImpl{ get_self<implementation::Command>(conflictingCmd) };
+                    const auto& conflictingCmdCopy{ conflictingCmdImpl->Copy() };
+                    conflictingCmdCopy->EraseKey(keys);
+                    _ConsolidatedActions.emplace(conflictingActionID, *conflictingCmdCopy);
+                }
+            }
+
+            // Assign the new action.
+            const auto actionID{ ActionHash{}(cmd.ActionAndArgs()) };
+            _KeyMap.insert_or_assign(keys, actionID);
+
+            if (oldCmd)
+            {
+                // Update inner Command with new key chord
+                auto oldCmdImpl{ get_self<Command>(oldCmd) };
+                oldCmdImpl->RegisterKey(keys);
+            }
+
+            if (consolidatedCmd)
+            {
+                // Update inner Command with new key chord
+                auto consolidatedCmdImpl{ get_self<Command>(consolidatedCmd) };
+                consolidatedCmdImpl->RegisterKey(keys);
+            }
+        }
+    }
+
+    // Method Description:
+    // - Retrieves the assigned command that can be invoked with the given key chord
+    // Arguments:
+    // - keys: the key chord of the command to search for
+    // Return Value:
+    // - the command with the given key chord
+    // - nullptr if the key chord is explicitly unbound
+    Model::Command ActionMap::GetActionByKeyChord(Control::KeyChord const& keys) const
+    {
+        // Check the current layer
+        const auto& cmd{ _GetActionByKeyChordInternal(keys) };

```

Comment:
`auto`

Reply:
Oh it might be interesting to mention why it needs to be `auto`!
As I mentioned a couple days ago you can't take the reference to the return value of a "prvalue", which is the value class of the return value of `call()` expressions for instance. In this language it's not possible to get a pointer to something that isn't on the stack / in memory yet, so the compiler has to put it on the stack first and then puts a pointer to that stack value into "cmd".

Oh and btw: Since it's a const value any `std::move` calls will _silently_ fail. Moving stuff requires the source to be mutable, since the source is "cleared out". `std::move` will still silently work, because a `reference&&` can decay to a `reference&` and thus create a regular copy.

It's such an innocent mistake... I'll make sure to keep an eye out for this in my reviews in the future.